File: 5-Problem-A-Day/45. Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        if (n < 2)
            return 0;
        int currMax = 0, nextMax = 0, i = 0, level = 0;
        while (currMax - i + 1 > 0) {
            level += 1;
            for (; i <= currMax; i++) {
                nextMax = Math.max(nextMax, nums[i] + i);
                if (nextMax >= n - 1)
                    return level;
            }
            currMax = nextMax;
        }
        return 0;
    }
}

File: 2023/45. Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if (nums.size() == 1) {
            return 0;
        }
        int step = 1, last = nums[0], reach = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (i > reach) {
                step++;
                reach = last;
            }
            last = max(i + nums[i], last);
        }
        return step;
    }
};